src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -294,9 +296,11 @@ class ClusterMonitorActor(val cluster: Cluster,
               // Otherwise, when a cluster is resumed and transitions from Starting to Running, we get stuck
               // in that state - at least with the way HttpJupyterDAO.isProxyAvailable works
               dbRef.inTransaction { _.clusterQuery.updateClusterHostIp(cluster.id, Some(ip)) }.flatMap { _ =>
-                jupyterProxyDAO.isProxyAvailable(cluster.googleProject, cluster.clusterName).map {
-                  case true => ReadyCluster(ip, googleInstances)
-                  case false => NotReadyCluster(ClusterStatus.Running, googleInstances)
+                dbRef.inTransaction { _.clusterImageQuery.getAllForCluster(cluster.id)}.flatMap { images =>
```

Comment:
if you turn this `flatMap` into a `map`, you won't need to wrap your resulting values in a `Future.successful()`

for a bit more explanation: 

you can think of `flatMap` as taking the result of the previous operation, pulling it out of the box (in this case, the box is a Future), and then running the following code without putting it back inside of the box. because your result isn't getting put in the box again, you have to do it manually by wrapping the result with `Future.successful()`

`map` works similarly, but it will put the result back inside of the box. so if you switch the above to a `map`, you don't have to manually wrap it in a `Future.successful()`.

that aside, for what i believe to be a more Scala-idiomatic way of writing your if/else, I would use `Future.traverse()`. you could do something like:

```
Future.traverse(images) { image => isProxyAvailable(image.tool) }.map { isAvailable =>
    if(isAvailable.forall(_ == true)) ReadyCluster(ip, googleInstances)
    else NotReadyCluster(ClusterStatus.Running, googleInstances)
}
```

note that this is using map, so there's no need to wrap the results in `Future.successful()`